src/c_clause/api/QAHandler.cpp

```diff
@@ -144,6 +178,45 @@ std::vector<std::vector<std::pair<int, double>>> QAHandler::getIdxAnswers(){
 }
 
 
+void QAHandler::writeAnswers(std::string outputPath, bool strings){
```

Comment:
remove getIdxAnswers here.  You are using this->answers below. Also we should only use it for serializing data to python as it copies (we may have to change this)